### PR TITLE
Update styled component dependency

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
updated `jest-styled-components` `styled-components` `ts-jest`
#### Where should the reviewer start?
package.json
#### What testing has been done on this PR?
yarn build-storybook
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
 backwards compatible

### CODE REVIEW

**Comment by britt6612** [comment]
Need to go through all snapshot changes to make sure they are correct due to upgrade

**Review by halocline — COMMENTED** [review]
Chromatic snapshots and changes to Storybook stories (e.g. /InfiniteScroll/Grid, /DataTable/MultiplePinned) seem concerning.

**Comment by britt6612** [comment]
Snap shot changes look correct I found that the there was a removal of the `styled-classnames`
This [PR](https://github.com/styled-components/jest-styled-components/pull/382) improves cleanup, so that style[data-styled-version] nodes are removed from the document.
```
the style rules are included in the snapshot, and the hashed class names are substituted with placeholders that make the diffs less noisy:
```

There also are new snapshot changes that seem to be from this [fix](https://github.com/styled-components/jest-styled-components/pull/319)

**Review by jcfilben — APPROVED** [review]
This looks good to me

File: src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap [hunk 1]
```diff
@@ -1615,27 +1721,33 @@ exports[`Layer invoke onClickOutside when modal={false} and layer has target 1`]
...
 @media only screen and (max-width: 768px) {
-  .dNkgGB {
+  .dbmsMO {
     position: relative;
     max-height: none;
     max-width: none;
     border-radius: 0;
-    height: 100vh;
-    width: 100vw;
+    height: 100%;
```
Comment (taysea) [inline src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap:1749]: This is switching from `100vh` to `100%` -- is this desired? or from a previous change that might not have received a snapshot update?

File: src/js/components/RadioButtonGroup/__tests__/__snapshots__/RadioButtonGroup-test.js.snap [hunk 2]
```diff
@@ -394,7 +394,7 @@ exports[`RadioButtonGroup boolean options 1`] = `
         </div>
       </div>
       <span
-        class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
+        class=""
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/RadioButtonGroup/__tests__/__snapshots__/RadioButtonGroup-test.js.snap:397]: noticing this in a few of the components and just confirming that is desired

[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/RadioButtonGroup/__tests__/__snapshots__/RadioButtonGroup-test.js.snap:397]: I noticed this when I looked through and I think it is happening in cases where there aren't any css properties associated with the class.

**Pull request closed without merging**